### ISSUE TITLE
Add  status enum for status codes

### ISSUE THREAD

**Issue by paulcsmith** [luckyframework/web#222]
They can be used anywhere status is used. For example `redirect`, `head` and `json`

```crystal
# Example usage
redirect to: Dashboard::Show, status: Status::Permanent
# But cleaner to use without `status:`
redirect to: Dashboard::Show, Status::Permanent
head Status::Created
# Also consider using an external name of `with`
redirect to: Dashboard::Show, with: Status::Permanent
```

This will make it impossible to have typos or an incorrect status. Also makes it easy to discover status because they'll show up in documentation and editor tooling

**Comment by hanneskaeufler** [comment]
Check out #232 for my take on this.

**Comment by paulcsmith** [comment]
Closed by #232
